gm4/plugins/readme_generator.py

```diff
@@ -0,0 +1,189 @@
+        })
+        if running_readme_gen and (v:=manifest_m_entry.get('pmc_link')):
+            site_replacements["pmc"].update({
+                f"\\(.+\\)<!--\\$dynamicLink:{m}-->": f"(https://planetminecraft.com/data-pack/{v})"
+        })
+
+    # Modrinth Youtube Video Embed
+    if (vid_url:=ctx.meta['gm4']['video']) is not None:
+        embed_url = re.sub(r'https:\/\/www.youtube.com\/watch\?v=(\w+)', r'https://www.youtube.com/embed/\1', vid_url)
+
+        site_replacements["modrinth"].update({
+            r"(.+)<!-- *\$modrinth:replaceWithVideo[ -]+?>" : (
+                "<iframe\n"
+                "width=\"640\"\n"
+                "height=\"480\"\n"
+                f"src=\"{embed_url}\"\n"
+                "frameborder=\"0\"\n"
+                "allow=\"autoplay; encrypted-media\"\n"
+                "allowfullscreen\n"
+                "></iframe>"
+            )
+        })
+
+    if running_readme_gen:
+        # PMC Element Deletion
+        site_replacements['pmc'].update({
+            r".+<!--\$pmc:delete.+>\n\n": ""
+        })
+
+        # Table BBCode conversion
+            # tables are gross to autodetect and capture, so we'll cheat with helper tags
+        tables = re.findall(r'<!-- *\$pmc:startTable ?-->\n((?:.+\n)+)<!-- *\$pmc:endTable ?-->', global_contents)
+        for table in tables:
+            repl = "[table]\n"
+            for line in table.strip('\n').split('\n'):
+                if set(line).issubset(set("| -:")):
+                    continue
+                repl += "[tr]\n"
+                for value in [v for v in line.split('|') if v != ""]:
+                    repl += f"\t[td]{value.strip()}[/td]\n"
+                repl += "[/tr]"
+            repl += "[/table]"
+
+            site_replacements['pmc'].update({
+                f'<!-- *\\$pmc:startTable ?-->\n{re.escape(table)}<!-- *\\$pmc:endTable ?-->':
+                    repl
+            })
+
+        # static BBCode translations
+        site_replacements['pmc'].update({
+            r"(.+)<!-- *\$pmc:headerSize.+>": r"[size=14px]\1[/size]",
+            r"#{2,3} (.+)": r"[style b size=14px]\1[/style]",
+            r"!\[(?!size|style|img|url)(.+)\]\((.+)\)": r"[img=\1]\2[/img]",
+            r"\[(?!size|style|img|url)(.+)\]\((.+)\)": r"[url=\2]\1[/url]",
+            r"<img src=\"(.+?)\" alt=(\".+?\") width=\"(.+?)\".+>": r"[img title=\2 width=\3]\1[/img]",
+            r"\*\*(.+)\*\*": r"[b]\1[/b]",
+            r"\*(.+)\*": r"[i]\1[/i]",
+            r"_([^_\n]+?)_(?![\w])": r"[i]\1[/i]",
+            r"__(.+)__": r"[u]\1[/u]",
+            r"~~(.+)~~": r"[s]\1[/s]",
+            r"`(.+)`": r"\1", # BBCode has no inline code blocks
+            r"```(.+)```": r"[code]\1[/code]",
+            r"---*\n": r"[hr]"
+        })
+
+        site_replacements['pmc'].update({
+            r"\n?<!--.+?-->": ""
```

Comment:
Why is this not together with the pmc replacements above?

Reply:
I swear there was a reason to separately remove lingering comments... but I can't find it now.